### PR TITLE
chore: updating RN default styling and aligning Button and Tabs primitives

### PR DESCRIPTION
**Pull request by dbanksdesign**
#### Description of changes

* Added `borderWidths` to RN base tokens
* Updated type for `fontWeights` to not have to cast in RN
* Updated the default `spaceModifier` to 1 and fixing padding sizes on primitives to match what exist on web
* Updated the variants of Button to be `primary`, `default`, and `link` to match RW and updated Authenticator screens to use the right variants
* Updated the Tabs primitive to match RW

#### Description of how you validated changes

| | | |
|-|-|-|
|Simulator Screen Shot - iPhone 11 - 2022-11-18 at 12 34 56 | Simulator Screen Shot - iPhone 11 - 2022-11-18 at 12 28 14 | Simulator Screen Shot - iPhone 11 - 2022-11-18 at 12 26 37
| Simulator Screen Shot - iPhone 11 - 2022-11-18 at 12 26 29 | Simulator Screen Shot - iPhone 11 - 2022-11-18 at 12 26 22

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-native/src/primitives/Button/Button.tsx [hunk 1]
```diff
@@ -36,16 +36,17 @@ export default function Button({
         disabled ? themedStyle.disabled : null,
         pressed ? themedStyle.pressed : null,
         pressedStateStyle,
-      ];
+      ] as StyleProp<ViewStyle>;
```
Comment (calebpollman) [inline packages/react-native/src/primitives/Button/Button.tsx:39]: Why do we need this cast now?

Comment (dbanksdesign) [reply packages/react-native/src/primitives/Button/Button.tsx:39]: B/c I touched this file so eslint gave me issues about implicit `any`s :/ Is there a better way to handle this?

File: packages/react-native/src/primitives/Tabs/styles.ts [hunk 2]
```diff
@@ -1,14 +1,30 @@
...
+
+  if (indicatorPosition && indicatorPosition === 'top') {
+    selectedTabBorderStyles.borderTopColor = colors.brand.primary[80];
+    tabBorderStyles.borderTopWidth = borderWidths.medium;
+    tabBorderStyles.borderTopColor = colors.border.secondary;
+  } else {
+    selectedTabBorderStyles.borderBottomColor = colors.brand.primary[80];
+    tabBorderStyles.borderBottomWidth = borderWidths.medium;
+    tabBorderStyles.borderBottomColor = colors.border.secondary;
+  }
```
Comment (calebpollman) [inline packages/react-native/src/primitives/Tabs/styles.ts:26]: Can be addressed outside of this PR but a little iffy on adopting this pattern in to component styling

File: packages/react-native/src/primitives/Label/styles.ts [hunk 3]
```diff
@@ -1,20 +1,18 @@
...
+    tokens: { colors, fontSizes, fontWeights },
   } = theme;
 
   return StyleSheet.create({
     text: {
       fontSize: fontSizes.medium,
-      fontWeight: fontWeights.normal as TextStyle['fontWeight'],
-      marginVertical: space.xxs,
-      marginHorizontal: space.xs,
+      fontWeight: fontWeights.normal,
```
Comment (calebpollman) [inline packages/react-native/src/primitives/Label/styles.ts:15]: ❤️

**Review by calebpollman — APPROVED** [review]
🚢

File: packages/react-native/src/primitives/TextField/styles.ts [hunk 4]
```diff
@@ -21,9 +21,9 @@ export const getThemedStyles = (theme: StrictTheme): TextFieldStyles => {
     },
     inputContainer: {
       alignItems: 'center',
-      borderColor: colors.black,
+      borderColor: colors.border.primary,
       borderRadius: radii.small,
-      borderWidth: 1,
+      borderWidth: borderWidths.small,
```
Comment (joebuono) [inline packages/react-native/src/primitives/TextField/styles.ts:26]: 👍

**Review by joebuono — APPROVED** [review]
🚢

**Pull request merged**